### PR TITLE
Fix fuzzy date trigger bug #32

### PR DESCRIPTION
**Pull request by thesved**
Fix for keyboard layouts where the guard character takes multiple keystrokes, eg. "alt + ,"

See: #32

### CODE REVIEW

**Comment by Stvad** [comment]
I saw you comments on `setTimeout` commit - I think just using `keyup` here without using any additional logic should work just as well here. can you verify?

**Comment by thesved** [comment]
Unfortunately not. For multi-key strokes `keyup` won't work, because if I hit `alt + ,` in a normal speed, and the current input is `;today`, this is what will happen:
1) `keydown: ;` - input is `;today`
2) `keypress: ;` - input is `;today`
3) `keyup: alt` - input is `;today;` <- `;` is arrived
4) `keyup: ,` - input is `;today;`

So with `keyup` there is no guarantee that you catch the intended character, see the difference between the semicolon in step 1) & 2), and the colon at step 4). 

At step 1) and 2) with `keydown` and `keypress` the problem is that the input is **not yet changed**, so this is why I did these 2 changes:
a) change `keyup` to `keypress`, so the key is released, but the input is not yet changed
b) defer the callback to the end of the event queue by `setTimeout`

Or the other solution could be to separately track the last pressed key, and base the trigger on the last pressed key, and not the current `keyup` event, but I thought that much more hackey.

If you put 3 different eventListeners to the document for `keydown`, `keypress`, and `keyup`, and log what key does the system get, you can play around with multi-stroke characters.

**Comment by Stvad** [comment]
I see, can you add explanation about the timeout as a comment in the code?

File: src/ts/contentScripts/dispatcher/index.ts [hunk 1]
```diff
@@ -20,6 +20,6 @@ const dispatchMap = new Map([
 
 browser.runtime.onMessage.addListener(command => dispatchMap.get(command)?.())
 
-document.addEventListener('keyup', ev => {
-    if (ev.key === guard) replaceFuzzyDate()
+document.addEventListener('keypress', ev => {
+    if (ev.key === guard) setTimeout(function(){ replaceFuzzyDate() },0)
```
Comment (Stvad) [inline src/ts/contentScripts/dispatcher/index.ts:24]: nit: I don't believe you need to wrap `replaceFuzzyDate` into additional function, I think you can just do:
`setTimeout(replaceFuzzyDate,0)`

Comment (Stvad) [reply src/ts/contentScripts/dispatcher/index.ts:24]: ah, another nit is that there should be space after coma :p

Comment (thesved) [reply src/ts/contentScripts/dispatcher/index.ts]: you are right, fixed it 😉

**Pull request merged**